File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
float average(float x, float y);

int main(void)
{
    // Get text from user
    string text = get_string("Text: ");

    // Determine the amount of letters, words, and sentences
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    //printf("Letters: %i, Sentences: %i, Words: %i\n", letters, sentences, words);
    // L is the average number of letters per 100 words in the text
    float L = average(letters, words);
    //printf("L: %f\n", L);
    // S is the average number of sentences per 100 words in the text
    float S = average(sentences, words);
    //printf("S: %f\n", S);

    // Coleman-Liau index
    int index = round((0.0588 * L - 0.296 * S) - 15.8);

    // Print result
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}

// Count the number of letters
int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
    }
    return (letters);
}

// Count the number of words
int count_words(string text)
{
    int words = 1;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == ' ')
        {
            words += 1;
        }
    }
    return (words);
}

// Count the number of sentences
int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if ((text[i] == '.') || (text[i] == '!') || (text[i] == '?'))
        {
            sentences += 1;
        }
    }
    return (sentences);
}

// Calculate averages
float average(float x, float y)
{
    float average;
    average = (x / y) * 100;
    return (average);
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Get text from user
    text = get_string("Text: ")

    # Determine the amount of letters, words, and sentences
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # L is the average number of letters per 100 words in the text
    L = average(letters, words)
    # S is the average number of sentences per 100 words in the text
    S = average(sentences, words)

    # Coleman-Liau Index
    index = round(((0.0588 * L) - (0.296 * S) - 15.8))

    # Print result
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print("Grade {}".format(index))


def count_letters(text):
    """ Returns the number of letters in a text"""
    letters = 0
    for char in text:
        if char.isalpha():
            letters += 1
    return letters


def count_words(text):
    """ Returns the number of words in a text"""
    words = 1
    for char in text:
        if char == ' ':
            words += 1
    return words


def count_sentences(text):
    """ Returns the number of sentences in a text"""
    sentences = 0
    for char in text:
        if (char == '.') or (char == '!') or (char == '?'):
            sentences += 1
    return sentences


def average(x, y):
    """ Returns the average of two numbers"""
    average = (x / y) * 100
    return average


if __name__ == "__main__":
    main()
